intake_geopandas/geopandas.py

```diff
@@ -70,15 +76,36 @@ def __init__(self, urlpath, bbox=None,
         self._bbox = bbox
         self._geopandas_kwargs = geopandas_kwargs or {}
         self._dataframe = None
+        self.storage_options = storage_options or {}
 
         super().__init__(metadata=metadata)
 
     def _open_dataset(self):
         """
         Open dataset using geopandas and use pattern fields to set new columns.
         """
+        # new caching
+        #url = fsspec.open_local(self.urlpath, **self.storage_options)
```

Comment:
I didnt get fiona to understand `fsspec`. should we anyways pass `storage_options`?

Reply:
The output of open_local should be a string, a path to a local file; but it works *only* if you are using caching in the URL. This could be of the form "simplecache::s3://bucket/path" (with storage options as {"simplecache": , "s3":}) or "simplecache://bucket/path" (with storage options {"target_protocol": "s3", "target_options": {..}}). I am in favour of allowing the fsspec route, but as @ian-r-rose said, the API would need to be clear about what was happening.

I'm not sure what "options" you can give to the GDAL loader (but if fsspec were using them, you would presumably want different options there).